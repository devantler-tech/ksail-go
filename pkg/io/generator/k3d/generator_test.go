package k3dgenerator_test

import (
	"testing"

	"github.com/devantler-tech/ksail-go/internal/testutils"
	"github.com/devantler-tech/ksail-go/pkg/apis/cluster/v1alpha1"
	generator "github.com/devantler-tech/ksail-go/pkg/io/generator/k3d"
	yamlgenerator "github.com/devantler-tech/ksail-go/pkg/io/generator/yaml"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestMain(m *testing.M) { testutils.RunTestMainWithSnapshotCleanup(m) }

func TestGenerate(t *testing.T) {
	t.Parallel()

	gen := generator.NewK3dGenerator()
	cluster := &v1alpha1.Cluster{}
	result, err := gen.Generate(cluster, yamlgenerator.Options{})
	require.NoError(t, err)
	require.NotEmpty(t, result)
	snaps.MatchSnapshot(t, result)
}
