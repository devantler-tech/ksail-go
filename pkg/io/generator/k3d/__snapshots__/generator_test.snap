
[TestGenerate - 1]
apiVersion: k3d.io/v1alpha3
kind: Simple
kubeAPI: {}
metadata: {}
options:
  k3d:
    disableImageVolume: false
    disableLoadbalancer: false
    disableRollback: false
    loadbalancer: {}
    wait: false
  k3s: {}
  kubeconfig: {}
  runtime:
    HostPidMode: false
registries: {}

---
